doc/Language/compilation.pod6

```diff
@@ -0,0 +1,295 @@
+Module will be compiled and the preprocessed version is then loaded when necessary. In practice, Modules that have been
+written by the community will only need to be pre-compiled once by a user when they are 'installed', for example by a
+Module manager such as C<zef>. Then they can be C<use'd>  by a developer in her own program. The effect is to make C<Raku>
+top level programs run quickly.
+
+One of the great strengths of the C<Perl> family of languages was the ability to integrate a whole ecosystem of modules
+written by competent programmers into a small program. This strength was widely copied and is now the norm for all
+languages. C<Raku> takes integration even further, making it relatively easy for C<Raku> programs to incorporate system
+libraries written in other languages into C<Raku> programs, see L<Native Call|Language/nativecall>.
+
+The experience from C<Perl> and other languages is that the distributive nature of Modules generate several practical difficulties:
+=item a popular module may go through several iterations as the API gets improved, without a guarantee that there is
+backward compatibility. So, if a program relies on some specific function or return, then there has to be a way to
+specify the B<Version>.
+=item a module may have been written by Bob, a very competent programmer, who moves on in life, leaving the module unmaintained,
+so Alice takes over. This means that the same module, with the same name, and the same general API may have have two
+versions in the wild. Alternatively, two developers (eg., Alice and Bob) who initially cooperated on a module, then part company about its
+development. Consequently, it sometimes is necessary for there to be a way to define the B<Auth> of the module.
+=item a module may be enhanced over time and the maintainer keeps two versions uptodate, but with different APIs. So it is
+may be necessary to define the B<API> required.
+=item when developing a new program a developer may want to have the modules written by both Alice and Bob installed locally.
+So it is not possible simply to have only one version of a module with a single name installed.
+
+C<Raku> enables all of these possibilities, allowing for multiple versions, multiple authorities, and multiple APIs to be present
+installed and available locally. The way classes and modules can be accessed with specific attributes
+is explained L<elsewhere|Language/typesystem#Versioning_and_authorship>. This tutorial is about how C<Raku> handles these
+possibilities.
+
+=head1 Introduction
+
+Before considering the C<Raku> framework, let's have a look at how languages like C<Perl> or C<Python> handle module
+installation and loading.
+
+=begin code
+ACME::Foo::Bar -> ACME/Foo/Bar.pm
+os.path -> os/path.py
+=end code
+
+In those languages, module names have a 1:1 relation with file system paths.
+We simply replace the double colons with slashes and add a .pm
```

Comment:
... or .py
It's not that easy in Python. You need to have a __init__.py in os to be considered a module. That's irrelevant, but, well...

Reply:
This is now niner's text